lib/pages/saved_places.dart

```diff
@@ -0,0 +1,375 @@
+  LatLng _center;
+  TextEditingController textController = TextEditingController();
+
+  @override
+  void dispose() {
+    textController.dispose();
+    super.dispose();
+  }
+
+  final Map<String, IconData> icons = <String, IconData>{
+    'saved_place:home': Icons.home,
+    'saved_place:work': Icons.work,
+    'saved_place:fastfood': Icons.fastfood,
+    'saved_place:local_cafe': Icons.local_cafe,
+    'saved_place:map': Icons.map,
+  };
+
+  @override
+  void initState() {
+    super.initState();
+    _loadState();
+  }
+
+  void _loadState() {
+    final SavedLocationsBloc savedLocationsBloc =
+        SavedLocationsBloc.of(context);
+    final TrufiConfigurationMap map = TrufiConfiguration().map;
+    final List<TrufiLocation> data =
+        savedLocationsBloc.locations.reversed.toList();
+
+    _center = map.center;
+    
+    final List<String> descriptions = data.map((TrufiLocation location) {
+      return location.description;
+    }).toList();
+
+    if (!descriptions.contains('Home')) {
+      savedLocationsBloc.inAddLocation.add(TrufiLocation(
+          description: 'Home',
+          latitude: _center.latitude,
+          longitude: _center.longitude,
+          type: 'saved_place:home'));
+    }
+
+    if (!descriptions.contains('Work')) {
+      savedLocationsBloc.inAddLocation.add(TrufiLocation(
+          description: 'Work',
+          latitude: _center.latitude,
+          longitude: _center.longitude,
+          type: 'saved_place:work'));
+    }
+    setState(() {
+    });
+  }
+
+  @override
+  Widget build(BuildContext context) {
+    return Scaffold(
+      appBar: _buildAppBar(context),
+      body: _buildBody(context),
+      drawer: TrufiDrawer(SavedPlacesPage.route),
+    );
+  }
+
+  Widget _buildAppBar(BuildContext context) {
+    final TrufiLocalization localization =
+        TrufiLocalizations.of(context).localization;
+    return AppBar(title: Text(localization.menuYourPlaces()));
+  }
+
+  Widget _buildBody(BuildContext context) {
+    return Container(
+      padding: const EdgeInsets.all(16.0),
+      child: Column(
+        children: <Widget>[
+          Expanded(child: _savedPlaces()),
+          _addPlaceBttn(),
+        ],
+      ),
+    );
+  }
+
+  //
+
+  Widget _savedPlaces() {
+    final ThemeData theme = Theme.of(context);
+    final TrufiLocalization localization =
+        TrufiLocalizations.of(context).localization;
+    final SavedLocationsBloc savedLocationsBloc =
+        SavedLocationsBloc.of(context);
+    final List<TrufiLocation> data =
+        savedLocationsBloc.locations.reversed.toList();
+
+    final SliverList saveds = SliverList(
+      delegate: SliverChildBuilderDelegate(
+        (BuildContext context, int index) {
+          final TrufiLocation savedPlace = data[index];
+          return Container(
+            margin: const EdgeInsets.only(bottom: 5),
+            child: RaisedButton(
+              onPressed: () {
+                log('bttn');
```

Comment:
The main action of the button is missing. I would have expected this to lock in the route from my current location to this place.